components/auth/org.wso2.carbon.auth.client.registration.rest.api/src/main/java/org/wso2/carbon/auth/client/registration/rest/api/impl/RegisterApiServiceImpl.java

```diff
@@ -120,6 +124,9 @@ public Response registerApplication(RegistrationRequestDTO registrationRequest,
         } catch (ParseException e) {
             log.error("Error while parsing authorization header", e);
             return Response.status(Response.Status.BAD_REQUEST).entity(RestAPIUtil.getInternalServerErrorDTO()).build();
+        } catch (AuthException e) {
+            log.error("Error ", e);
```

Comment:
Best to give a better explanation as to the cause of the error. Due to GDPR compliance no need to mention the actual users name. Logging the pseudo name is sufficient for debugging purposes.